### ISSUE TITLE
Add CSV file to repo or modify code

### ISSUE THREAD

**Issue by TobiEnveda** [vfscalfani/CSN_tutorial#1]
The file glucocorticoid_receptor_2034_2.csv is missing from the repo? Maybe download from article and rename file, or upload the CSV to the Github?

### 2. Load ChEMBL Dataset

!wget https://static-content.springer.com/esm/art%3A10.1186%2Fs13321-022-00664-x/MediaObjects/13321_2022_664_MOESM1_ESM.csv
!mv 13321_2022_664_MOESM1_ESM.csv glucocorticoid_receptor_2034_2.csv

**Comment by vfscalfani** [comment]
Hi @TobiEnveda 

Many thanks for your interest and submitting the issue. Since I'm not sure how stable the Springer additional file link will be, I went ahead and added the dataset file to this repository. Thanks, hope you find the code and paper useful. 

Vin

**Comment by TobiEnveda** [comment]
Hi @vfscalfani,
thank you! I am still thinking about the limiting step the "tan_mcs"  "tan_similarity" calculations and how that could be speed up. Because as you mentioned "This may take a very long time..." However 81k calculations (81406) should be finished in seconds, not hours. Even without vectorization that should be extremely fast. 
Tobias

**Comment by vfscalfani** [comment]
Hi @TobiEnveda yes, I would have thought so too, let us know if you find anything we missed. 

Also, check out the modified "less memory" script:

https://github.com/vfscalfani/CSN_tutorial/blob/main/Less_Memory_Calculations/CSN_calculations_lessMem.py

The multiprocessing fingerprint calculations do take only seconds now (with single processor, it is a few minutes). For the maximum common substructure (MCS) calculations, these are indeed computationally intensive and can take up to 10 seconds per calculation (which is the max limit we set to return the best match in 10 seconds with the RDKit function). 

The only thing we have found that speeds up the MCS calculations is to throw more processors at it. The speed also greatly depends on the molecules being compared. So, if you have a different dataset with > 80,000 subsets, that does not necessarily mean the MCS calculations will be longer, it depends on the molecules being compared.

Vin
